### PR TITLE
Added doc for palindromic linked list

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/38

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Please make changes in the `sidebars.js`on the root. Otherwise it would be impossible to review the article on a live preview.

**Comment by SAY-droid427** [comment]
@HarshCasper Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+# Algorithm
+
+1. Create a linked list
+2. Initialize a stack
+3. Push the data of each node in the linked list into the stack till the end
+4. Initialize a pointer to the head of the linked list
+5. Initialize a temporary variable and store the data present at the top of the stack.
+   Remember that this data is actually the data of the last node of the linked list.
+6. Compare the value in the temporary variable with that of the node pointed to by the pointer in the linked list
+   Remember that the pointer points to the head , that is, the first node of the linke dlist.
```
Comment (vybhav72954) [inline docs/data-structures/palindromic-linked-list.mdx:21]: Please correct the spelling of Linked List here

File: docs/data-structures/palindromic-linked-list.mdx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+It is now compared with the top of the stack which is 3, so they are not equal and hence linked list is not palindrome.
+
+## Example 2
+
+Let the linked list be 1->2->1
+All the data is pushed into the stack 
+Compare the top of the stack(=1) with the data at head of the linked list.
+As they are equal, pop the current top of the stack and pointer points towards the next node(=2)
+Again both are equal. Repeat the above step again. 
+You have reached the end of the linked list. therefore, it is palindromic.
```
Comment (vybhav72954) [inline docs/data-structures/palindromic-linked-list.mdx:42]: Therefore should start with a capital letter.

File: docs/data-structures/palindromic-linked-list.mdx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+### Input:
+Enter the length of the list 3
+Enter the data
+1
+2
+1
+
+### Output:
+Yes,linked list is palindrome
```
Comment (vybhav72954) [inline docs/data-structures/palindromic-linked-list.mdx:138]: Interesting, your print statement has whitespace, never the less, pls add one here as well.

File: docs/data-structures/palindromic-linked-list.mdx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+### Input:
+Enter the length of the list 3
+Enter the data
+1
+2
+9
+
+### Output:
+No,linked list is not palindrome
```
Comment (vybhav72954) [inline docs/data-structures/palindromic-linked-list.mdx:150]: Here as well

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
@HarshCasper From a grammar point of view, after the mentioned changes, this looks good.

File: docs/data-structures/palindromic-linked-list.mdx (hunk 1 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 2 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 3 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 4 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx]: Done

**Review by vybhav72954 — APPROVED** [review]

File: docs/data-structures/palindromic-linked-list.mdx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+1
+2
+1
+
+### Output:
+Yes, linked list is palindrome
+
+##Sample 2
+
+### Input:
```
Comment (HarshCasper) [inline docs/data-structures/palindromic-linked-list.mdx:142]: Keep heading 4

File: docs/data-structures/palindromic-linked-list.mdx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+Enter the length of the list: 3
+Enter the data:
+1
+2
+9
+
+### Output:
```
Comment (HarshCasper) [inline docs/data-structures/palindromic-linked-list.mdx:149]: Keep heading 4

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
Hi @SAY-droid427 

This was the last set of review. Kindly make the changes and I will merge it :tada: 

(Also please fix the merge conflicts)

File: docs/data-structures/palindromic-linked-list.mdx [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+8. If the end of the linked list is reached, then it means that the linked list is palindromic. Otherwise it is not palindromic.
+
+# Examples
+
+## Example 1
+
+Let the linked list be 1->2->3
+All the data is pushed into the stack in the order: 3 2 1 with 1 at the bottom and 3 at top
+The pointer points towards the head of the linked list that is 1
+It is now compared with the top of the stack which is 3, so they are not equal and hence linked list is not palindrome.
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:33]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+## Example 2
+
+Let the linked list be 1->2->1
+All the data is pushed into the stack 
+Compare the top of the stack(=1) with the data at head of the linked list.
+As they are equal, pop the current top of the stack and pointer points towards the next node(=2)
+Again both are equal. Repeat the above step again. 
+You have reached the end of the linked list. Therefore, it is palindromic.
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:42]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+1
+2
+9
+
+### Output:
+No, linked list is not palindrome
+
+# Complexity Analysis
+
+* Time Complexity : O(n)
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:154]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+* Space complexity : O(n)
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:155]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+    cout<< "No, linked list is not palindrome";
+}
+
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+## Sample 1
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:128]: Should I remove Sample 1 heading?

File: docs/data-structures/palindromic-linked-list.mdx [hunk 12]
```diff
@@ -0,0 +1,158 @@
+---
+title: Palindromic Linked List 
+slug: /data-structures/palindromic-linked-list
+---
+
+# Introduction
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:6]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+A palindrome is a word or combination of characters that read the same forward and backward.
+Whether a linked list is palindromic or not can be found out using stacks - a data structure. Stacks follow the rule of 'Last In First Out',
+popularly known as 'LIFO'.
+
+# Algorithm
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:12]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+4. Initialize a pointer to the head of the linked list
+5. Initialize a temporary variable and store the data present at the top of the stack.
+   Remember that this data is actually the data of the last node of the linked list.
+6. Compare the value in the temporary variable with that of the node pointed to by the pointer in the linked list
+   Remember that the pointer points to the head , that is, the first node of the linked list.
+7. If the two values are the same, then move onto the next node in the linked list and pop the current top element in the stack.
+   Repeat step 6 till the end of the stack or linked list is reached or the two values being compared are not the same.
+8. If the end of the linked list is reached, then it means that the linked list is palindromic. Otherwise it is not palindromic.
+
+# Examples
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:26]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+
+## Example 1
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:28]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+
+Let the linked list be 1->2->3
+All the data is pushed into the stack in the order: 3 2 1 with 1 at the bottom and 3 at top
+The pointer points towards the head of the linked list that is 1
+It is now compared with the top of the stack which is 3, so they are not equal and hence linked list is not palindrome.
+
+## Example 2
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:35]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 17]
```diff
@@ -0,0 +1,158 @@
...
+
+Let the linked list be 1->2->1
+All the data is pushed into the stack 
+Compare the top of the stack(=1) with the data at head of the linked list.
+As they are equal, pop the current top of the stack and pointer points towards the next node(=2)
+Again both are equal. Repeat the above step again. 
+You have reached the end of the linked list. Therefore, it is palindromic.
+
+
+# Code
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:45]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 18]
```diff
@@ -0,0 +1,158 @@
...
+    cout << "Yes, linked list is palindrome";
+    else if( checkPalin == -1)
+    cout<< "No, linked list is not palindrome";
+}
+
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:126]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+Enter the length of the list: 3
+Enter the data:
+1
+2
+9
+
+### Output:
+No, linked list is not palindrome
+
+# Complexity Analysis
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:152]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+## Sample 1
+
+### Input:
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:130]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 21]
```diff
@@ -0,0 +1,158 @@
...
+
+### Output:
+No, linked list is not palindrome
+
+# Complexity Analysis
+
+* Time Complexity : O(n)
+* Space complexity : O(n)
+
+# Credits
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:157]: Done

File: docs/data-structures/palindromic-linked-list.mdx [hunk 22]
```diff
@@ -0,0 +1,158 @@
...
+## Sample 1
+
+### Input:
+Enter the length of the list: 3
+Enter the data:
+1
+2
+1
+
+### Output:
```
[the remark this answers is not in the document]
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:137]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 5 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:142]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 6 above)
Comment (SAY-droid427) [reply docs/data-structures/palindromic-linked-list.mdx:149]: Done

File: docs/data-structures/palindromic-linked-list.mdx (hunk 11 above)
[the remark this answers is not in the document]
Comment (HarshCasper) [reply docs/data-structures/palindromic-linked-list.mdx:128]: Yes

**Comment by SAY-droid427** [comment]
@HarshCasper I have made all the changes and also learnt how to resolve the issue of hint: Updates were rejected because the tip of your current branch is behind

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**